providers/screensaver.rb

```diff
@@ -31,39 +31,31 @@
   ## Distinguir entre sesion Cinnamon y LXDE
 
   #     session = node["desktop_session"] 
-        gecos_ws_mgmt_desktop_setting "idle-activation-enabled" do
-          type "string"
+        desktop_gsettings "org.cinnamon.desktop.screensaver" do
+          key "idle-activation-enabled"
+          user username
           value idle_enabled.to_s
-          schema "org.cinnamon.desktop.screensaver"
-          username username
-          provider "gecos_ws_mgmt_gsettings"
           action :nothing
         end.run_action(:set)
     
-        gecos_ws_mgmt_desktop_setting "lock-enabled" do
-          type "string"
+        desktop_gsettings "org.cinnamon.desktop.screensaver" do
```

Comment:
Avoid reusing a resource name ("org.cinnamon.desktop.screensaver"). Append the key name to this resource.